src/logger.rs

```diff
@@ -1,33 +1,89 @@
+use crate::{log_configuration::LogConfiguration, Buffer, Priority, TagMode};
 #[cfg(target_os = "android")]
 use crate::{thread, Record};
-use crate::{Buffer, Priority, TagMode};
 use env_logger::filter::Filter;
 use log::{LevelFilter, Log, Metadata};
 use std::io;
+use std::sync::{Arc, RwLock};
 #[cfg(target_os = "android")]
 use std::time::SystemTime;
 
-pub(crate) struct Logger {
-    filter: Filter,
-    tag: TagMode,
-    prepend_module: bool,
-    #[allow(unused)]
-    pstore: bool,
-    #[allow(unused)]
-    buffer_id: Buffer,
+///Logger configuration handler stores access to logger configuration parameters
+///
+///
+#[derive(Clone)]
+pub struct LoggerConfigHandler {
+    pub(crate) configuration_handler: Arc<RwLock<LogConfiguration>>,
+}
+
+impl LoggerConfigHandler {
+    /// Create new configuration handler from Arc<RwLock<LogConfiguration>>
+    ///
+    pub(crate) fn new(configuration: Arc<RwLock<LogConfiguration>>) -> Self {
+        Self {
+            configuration_handler: configuration,
+        }
+    }
+
+    /// Create new configuration handler from LogConfiguration object
+    ///     
+    pub(crate) fn new_from_raw(configuration: LogConfiguration) -> Self {
+        Self {
+            configuration_handler: { Arc::new(RwLock::new(configuration)) },
+        }
+    }
+
+    /// Provide access to configuration parameters for changing
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use log::LevelFilter;
+    /// # use android_logd_logger::Builder;
+    ///
+    /// let config = android_logd_logger::builder()
+    /// .parse_filters("debug")
+    /// .init();
+    ///
+    /// config.setter().set_level_filter(LevelFilter::Error);
+    /// ```
+    pub fn setter(&self) -> std::sync::RwLockWriteGuard<LogConfiguration> {
+        self.configuration_handler.write().unwrap()
+    }
+
+    /// Provide access to configuration parameters for reading
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use log::LevelFilter;
+    /// # use android_logd_logger::Builder;
+    ///
+    /// let config = android_logd_logger::builder()
+    /// .parse_filters("debug")
+    /// .init();
+    ///
+    /// let prepend_module = config.getter().get_prepend_module();
+    /// ```
+    pub fn getter(&self) -> std::sync::RwLockReadGuard<LogConfiguration> {
+        self.configuration_handler.read().unwrap()
+    }
```

Comment:
Not sure if we need "getter" at all.

Reply:
Implementation LoggerImpl for Log contain a lot of places where need read fields, and for my opinion getter is most useful tool in this case